### PR TITLE
Feature/14 auth service for client

### PR DESCRIPTION
**Pull request by Limbo2332**

### CODE REVIEW

File: frontend/src/app/core/services/auth.service.ts [hunk 1]
```diff
@@ -15,33 +16,21 @@ export class AuthService {
 
     private userNameInLS = 'userInfo';
 
-    constructor(private httpClient: HttpClient) {
-        // do nothing
-    }
+    constructor(private httpClient: HttpClient, private httpService: HttpInternalService) {}
```
Comment (tatianahutii) [inline frontend/src/app/core/services/auth.service.ts:19]: Do we still need httpClient here?

**Review by tatianahutii — CHANGES_REQUESTED** [review]

File: frontend/src/app/core/services/auth.service.ts (hunk 1 above)
Comment (Limbo2332) [reply frontend/src/app/core/services/auth.service.ts]: Yeah, I deleted, but some checks were not successful

File: frontend/src/app/core/services/auth.service.ts [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    private userSubject: BehaviorSubject<FakeUser | undefined>;
+
+    private user: FakeUser | undefined;
+
+    private userNameInLS = 'userInfo';
+
+    constructor(private httpService: HttpInternalService) {}
+
+    public isAuthorized() {
+        return this.user !== undefined && this.getUserInfo() !== undefined;
```
Comment (tatianahutii) [inline frontend/src/app/core/services/auth.service.ts:21]:
```suggestion
        return this.user  && this.getUserInfo()
```

**Review by tatianahutii — APPROVED** [review]
LGTM

**Pull request merged**